physionet-django/console/views.py

```diff
@@ -716,6 +699,48 @@ def manage_published_project(request, project_slug, version):
          'rw_tasks': rw_tasks, 'ro_tasks': ro_tasks,
          'anonymous_url':anonymous_url, 'passphrase':passphrase})
 
+def gcp_bucket_management(request, project, user):
+    """
+    Create the database object and cloud bucket if they do not exist, and send
+    the files to the bucket.
+    """
+    is_private = True
+
+    if project.access_policy == 0:
+        is_private = False
+        group = None
+    else:
+        group = utility.get_bucket_email(project.slug, project.version)
+
+    bucket_name = utility.get_bucket_name(project.slug, project.version)
+
+    if GCP.objects.filter(bucket_name=bucket_name):
+        messages.success(request, "The bucket already exists. Resending the \
+            files for the project {0}.".format(project))
+    else:
+        if utility.check_bucket_exists(project.slug, project.version):
+            LOGGER.info("The bucket {0} already exists, skipping bucket and \
```

Comment:
`GCP.objects.create` and the `if group` clause both appear after `else`, so the logger message doesn't appear to be true.

Reply:
If the object already exists, there is no reason to re-enter to the group creation section. I don't understand what you want to say in this comment.

Reply:
Looking at this in more detail, I now understand. I'd missed that `create_bucket` was also creating the access group, and had thought that "skipping group creation" referred to the `DataAccess.objects.create` step (that is not being skipped.